.circleci/config.yml

```diff
@@ -74,39 +74,46 @@ jobs:
       - checkout
       - go/mod-download-cached
       - setup_remote_docker:
-          version: 17.07.0-ce
+          version: 18.06.0-ce
       - attach_workspace:
           at: .
-      - run: ln -s .build/linux-amd64/thanos thanos
-      - run: make docker
-      - run: docker run thanos --help
+      # Register qemu to support multi-arch.
+      - run: docker run --privileged linuxkit/binfmt:v0.7
+      - run: make crossbuild
```

Comment:
Last improvement that I'd like to see before merging: this builds the binary for all supported archs/OSes but for `main` we only care about two - arm64/amd64. Could we only build those? It takes quite a long time to build for all OSes on my laptop so I can only imagine what it will be like on CircleCI :smile:

Reply:
I have added the following logic: if on the main branch, only build above two.